### ISSUE TITLE
Accuracy/Loss plots are buggy for multiple validation sets

### ISSUE THREAD

**Issue by Saphereye** [tracel-ai/burn#1496]
**Describe the bug**
When utilizing the CLI tool, an anomaly occurs where, upon plotting the Accuracy and Loss data in a line plot, the plot contains extraneous lines connecting them when more than two sets are present.

**To Reproduce** 
 1. Create a repo for the [Basic workflow example](https://burn.dev/book/basic-workflow/index.html)
 2. Run the program (with the CLI version) and Wgpu device.
 3. Wait for two or more validation set graphs

**Expected behavior**
Although uncertain if this is intended functionality, the lines connecting validation data from distinct epochs should not occur.

**Screenshots**
28% progress Accuracy plot, two validation sets covered
![Screenshot from 2024-03-21 17-00-15](https://github.com/tracel-ai/burn/assets/59739923/db0d97e2-675f-4056-998b-da1929dc4485)

29% progress Loss plot after two validation sets
![Screenshot from 2024-03-21 17-00-40](https://github.com/tracel-ai/burn/assets/59739923/5908bb16-6531-46f0-8066-c953541d0c1e)

37% progress Accuracy plot, three validation sets covered
![image](https://github.com/tracel-ai/burn/assets/59739923/df3e5ba9-81bd-4752-8efa-27597ab8b876)

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.4 LTS x86_64
 - Burn: **Version** 0.12.1, **Features** train, wgpu, vision

**Additional context**
I am following the Burn [Basic workflow example](https://burn.dev/book/basic-workflow/index.html) for this example.

I believe the data for the plots is being updated [here](https://github.com/tracel-ai/burn/blob/main/crates/burn-train/src/renderer/tui/renderer.rs). A possible solution could be to draw new lines for each epoch.

Your dedication to investigating this matter is greatly appreciated. Thank you for your valuable contributions to this project.

**Comment by louisfd** [comment]
It's true that it's not very beautiful. If you have an idea of how to separate the blocks with a solution that is not too hacky, then maybe you can try and contribute? 😄

**Comment by Saphereye** [comment]
@louisfd I think I worded the issue incorrectly. Beautiful graphs and data on TUIs are what I had envisioned when I had initially started with computers, and this visualisation method scratches that itch :heart:

I would love to contribute to the project!

**Comment by antimora** [comment]
The TUI renderer changed a lot since 0.12. Could you retest on current main (or 0.22 once it's out)? If we don't hear back in a few weeks we'll assume it's fixed and close this.
